### PR TITLE
Capic

### PR DESCRIPTION
**Pull request by jojimt**
First cut changes for capic - opflex-overlay integration.

@snaiksat @apoorva11029

### CODE REVIEW

File: provision/acc_provision/templates/aci-containers.yaml [hunk 1]
```diff
@@ -235,14 +242,22 @@ data:
         "aci-vrf": {{config.aci_config.vrf.name|json}},
         "default-endpoint-group": {
             "policy-space": {{config.kube_config.default_endpoint_group.tenant|json}},
+        {% if config.flavor == "eks" %}
```
Comment (snaiksat) [inline provision/acc_provision/templates/aci-containers.yaml]: do this and some of the other changes below also apply to the localhost flavor?

File: provision/acc_provision/apic_provision.py [hunk 2]
```diff
@@ -76,21 +88,26 @@ def __init__(
         if self.cookies is None:
             self.login()
             if self.cookies is not None:
-                apic_cookies[(addr, username, ssl)] = self.cookies
+                #apic_cookies[(addr, username, ssl)] = self.cookies
+                print("apic_cookies {}".format(apic_cookies))
+                print("self {}".format(self.cookies))
+
```
Comment (snaiksat) [inline provision/acc_provision/apic_provision.py]: revert this change?

File: provision/acc_provision/acc_provision.py [hunk 3]
```diff
@@ -1093,10 +1128,14 @@ def provision(args, apic_file, no_random):
             err("Not able to login to the APIC, please check username or password")
             return False
 
+    print("Logged in")
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py]: needed?

File: provision/acc_provision/acc_provision.py [hunk 4]
```diff
@@ -528,48 +543,60 @@ def Raise(exception):
...
-                                    required),
-        "net_config/node_subnet": (get(("net_config", "node_subnet")),
-                                   required),
         "net_config/pod_subnet": (get(("net_config", "pod_subnet")),
                                   required),
-        "net_config/extern_dynamic": (get(("net_config", "extern_dynamic")),
+    }
+
+    if isOverlay(config["flavor"]):
+        print("Got overlay")
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py]: nit: perhaps say - "Using overlay"

File: provision/acc_provision/flavors.yaml [hunk 5]
```diff
@@ -262,6 +262,7 @@ flavors:
   eks:
     desc:         Kubernetes EKS
     default_version: 4.2
+    overlay: True
```
[the remark this answers is not in the document]
Comment (jojimt) [reply provision/acc_provision/flavors.yaml]: As of now, this is only needed when cAPIC is involved.

File: provision/acc_provision/templates/aci-containers.yaml (hunk 1 above)
Comment (jojimt) [reply provision/acc_provision/templates/aci-containers.yaml]: As of now, no.

File: provision/acc_provision/acc_provision.py [hunk 6]
```diff
@@ -918,6 +960,7 @@ def get_apic(config):
     apic_host = config["aci_config"]["apic_hosts"][0]
     apic_username = config["aci_config"]["apic_login"]["username"]
     apic_password = config["aci_config"]["apic_login"]["password"]
+    print(apic_password)
```
Comment (apoorva11029) [inline provision/acc_provision/acc_provision.py]: Delete this?
